### DISCUSSION TITLE
6.0.0

### DISCUSSION THREAD

**Discussion by XilinJia** [XilinJia/Podcini#58] [Announcements]
* complete overhaul of database and routines, ditched the iron-age celebrity SQLite and entrusted the modern object-based Realm
* export/import DB is supported for the Realm DB with file extension .realm
* DB from Podcini version 5 and below can not be imported, see migrationTo6.md file on github for instructions
* deleting feeds is performed promptly
* components rely more on objects for communication, unnecessary DB access is reduced
* subscriptions sorting is more explicit
* in Subscriptions view, click on cover image of a feed opens the FeedInfo view (not FeedEpisodes view)
* in Subscriptions and episodes list views, corrected the issue of wrong images of episodes being shown when scrolling
* the Counter and its seetings are removed
* flow event additions and improvements
* receiving flow events are strictly tied to life cycles of the components
* NavDrawer no longer gets updated in the background but only upon open
* player control UI is more efficient
* PlaybackController is further enhanced for multiple access
* non-essential instantiations of PlaybackController are stripped
* AudioPlayer view is hidden when there is no media set to play
* playback routines are extensively tuned and cleaned, less layered
* in any episode list views, swipe with NO_ACTION defined pops up the swipe config dialog
* episodes marked played will be removed from all queues
* in EpisodeInfo view, "mark played/unplayed", "add to/remove from queue", and "favoraite/unfovorite" are at the action bar
* decade-old joanzapata iconify is replaced with mikepenz iconics
* removed the need for support libraries and the need for the jetifier
* Java tools checkstyle and spotbus are removed
* the clumsy FeedDrawerItem class was removed and related compponents are based purely on feed objects
* FeedInfo view has button showing number of episodes to open the FeedEpisodes view
* on action bar of FeedEpisodes view there is a direct access to Queue
* tidied up the header of FeedEpisodes view
* media size is shown on episode info view
* net-fetching of media size for not-downloaded media is removed for episode list views
* there is a setting to disable/enable auto backup OPML files to Google
* all RxJava code was replaced with coroutines, RxJava dependency is kept only for using fyyd search
* PlaybackPreferences using SharePreferences was removed and related info is handled by the DB as CurrentState
* InTheatre object is now the center reference for all currently playing stuff including the current play queue
* 5 queues are provided by default: Default queue, and Queues 1-4
	* all queue operations are on the curQueue, which can be set in all episodes list views
	* on app startup, the most recently updated queue is set to curQueue
* extensive adjustments project class structures
* code is now built with Kotlin 2.0.0

<hr /><em>This discussion was created from the release <a href='https://github.com/XilinJia/Podcini/releases/tag/6.0.0'>6.0.0</a>.</em>

**Comment by summersab** [comment]
I've identified a number of bugs with the latest version of v6:
1. Player in the notification is not visible from the lockscreen. There is a space where the player SHOULD be, but it is invisible.
2. Toggling filters when an episode is playing causes the app to crash.
3. When refreshing a podcast that has unplayed episodes and the "Not played" filter enabled, the list of episodes shows duplicate entries of these unplayed episodes once the refresh is complete. Going back to another screen and then reopening the podcast subscription page clears these entries.
4. With the "Not played" filter enabled, marking an episode as played does not hide the entry until exiting the episode list and reopening it.
5. When episodes are in the queue, continuous playback is enabled, and playback begins with an episode other than the first entry of the queue, the queue begins playing from the beginning once the final episode is played. The default used to be to stop once the last episode is played. I often have long podcasts queued up for later, but I add short ones to the end and listen to them first while intending to leave the longer episodes for later.
6. The "Set navigation drawer items" setting is broken and the selection is neither applied nor saved.
7. After a crash, the app begins automatically playing whatever is in the current queue. Since the current release has a few bugs that cause crashes, this is pretty annoying.

I could open bug reports for these, but I imagine many of them are on your radar. So, I wanted to put them here first, and I will add new comments as necessary. Let me know what would be most helpful so you can track reports.

**Reply by XilinJia** [reply]
Thanks for reporting.
Are you using the most updated version or just 6.0.0?
On 1, what is your Android version? I have that on lockscreen on Android 14.
Can you explain more on 2 ?
I noticed 4.
Regarding 5, it's because the new queue is circular as mentioned in the changelog. Can you take advantage of multiple queues for your specific usage?

**Reply by XilinJia** [reply]
coming in the next release:
* in FeedEpisode view fixed filtering after an episode's play state is changed
* fixed refreshing a feed causes duplicate episodes in FeedEpisodes view
* fixed the non-functioning of "Set navigation drawer items"

**Reply by summersab** [reply]
1. I'm running Podcini 6.0.10, Android 14 (LOS 21-20240525-microG-guacamole)
2. If a podcast is playing, tap the filter button in the podcast subscription view, change the filter, and select "Confirm." It only happens if media is actually playing from what I can tell.
3. Fixed as noted.
4. Fixed as noted.
5. I haven't messed with multiple queues, yet, but I am eager to do so. I have wanted this feature for awhile but never requested it because I figured it would be hard to implement. So, thank you for doing it! However, maybe continuous/loop playback should be optional? I don't really mind either way - I'll get used to it.

**Reply by summersab** [reply]
I added a new bug to the OP above.

For point 2, it seems to require toggling the filters more than once (maybe 2-3 times). So, begin playing media, change and confirm filters a few times, and the app crashes.

For point 6, I'm going to TRY to get used to the loop behavior, but I'm already not a fan. I just encountered an issue where playback continues with multiple queues, but I'll need to do some more testing to provide an example.

**Comment by XilinJia** [comment]
coming in next release:
* fixed crash when changing filter during media playing
